docs/en/stack/ml/nlp/ml-nlp-autoscaling.asciidoc

```diff
@@ -0,0 +1,144 @@
+Refer to the tables in the <<auto-scaling-matrix>> section to find out the setings for the level you selected.
+
+
+[role="screenshot"]
+image::images/ml-nlp-deployment-id-elser-v2.png["ELSER deployment with adaptive resources enabled.",width=640]
+
+
+[discrete]
+[[auto-scaling-matrix]]
+== Model deployment resource matrix
+
+The used resources for trained model deployments depends on three factors:
+
+* your cluster environment (Serverless, Cloud, or on-premises)
+* the use case you optimize the model deployment (ingest or search)
+* whether adaptive resources are enabled or disabled (dynamic or static resources)
+
+If you use {es} on-premises, adaptive resources behavior is fully dynamic and highly dependent on the hardware configuration.
+The following tables show you the number of allocations, threads, and vCPUs available in Cloud when adaptive resources are enabled or disabled.
+
+
+[discrete]
+=== Deployments in Cloud optimized for ingest
+
+In case of ingest-optimized deployments, we maximize the number of model allocations.
+
+
+[discrete]
+==== Adaptive resources enabled
+
+[cols="4*", options="header"]
+|==========
+| Level  | Allocations                                  | Threads | vCPUs
+| Low    | 0 to 2 dynamically                           | 1       | 0 to 2 dynamically 
+| Medium | 1 to 32 dynamically                          | 1       | 1 to the smaller of 32 or the limit set in the Cloud console, dynamically
+| High   | 1 to limit set by Cloud console, dynamically | 1       | 1 to limit set by Cloud console, dynamically
+|==========
+
+
+[discrete]
+==== Adaptive resources disabled
+
+[cols="4*", options="header"]
+|==========
+| Level  | Allocations                                              | Threads | vCPUs
+| Low    | 2 if available, otherwise 1, statically                  | 1       | 2 if available
+| Medium | 32 if available, otherwise maximum available, statically | 1       | 32 if available
+| High   | Maximum available set by Cloud console, statically       | 1       | Maximum available set by Cloud console, statically
+|==========
+
+
+[discrete]
+=== Deployments in Cloud optimized for search
+
+In case of search-optimized deployments, we maximize the number of threads.
+The maximum number of threads that can be claimed depends on the hardware your architecture has.
+
+[discrete]
+==== Adaptive resources enabled
+
+[cols="4*", options="header"]
+|==========
+| Level  | Allocations                                  | Threads                                            | vCPUs
+| Low    |  1                                           | 2                                                  | 2
+| Medium |  1 to 2 (if threads=16) dynamically          | maximum that the hardware allows (for example, 16) | 1 to 32 dynamically
+| High   |  1 to limit set by Cloud console, dynamically| maximum that the hardware allows (for example, 16) | 1 to limit set by Cloud console, dynamically
```

Comment:
To be precise, you don't set an allocations limit in the cloud console, but a vCPU limit. The leads to an allocations limit of vCPU limit divided by the number of threads. Good luck phrasing that nicely!